contrib/go.mongodb.org/mongo-driver.v2/mongo/mongo_test.go

```diff
@@ -80,3 +81,68 @@ func Test(t *testing.T) {
 	assert.Equal(t, ext.SpanKindClient, s.Tag(ext.SpanKind))
 	assert.Equal(t, "mongodb", s.Tag(ext.DBSystem))
 }
+
+func TestTruncation(t *testing.T) {
+	getQuery := func(t *testing.T, max int) (string, bool) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+		defer cancel()
+
+		span, ctx := tracer.StartSpanFromContext(ctx, "mongodb-test")
+
+		addr := "mongodb://localhost:27017/?connect=direct"
+		opts := options.Client()
+		opts.Monitor = NewMonitor(WithMaxQuerySize(max))
+		opts.ApplyURI(addr)
+		client, err := mongo.Connect(opts)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = client.
+			Database("test-database").
+			Collection("test-collection").
+			UpdateOne(
+				ctx,
+				bson.D{{Key: "_id", Value: "68536ec8d906742797f5705a"}},
+				bson.D{{Key: "$set", Value: map[string]any{"test-item": "test-value"}}},
+			)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		span.Finish()
+
+		spans := mt.FinishedSpans()
+		value, ok := spans[0].Tag("mongodb.query").(string)
+		return value, ok
+	}
+
+	t.Run("zero", func(t *testing.T) {
+		// Should *not* attach the tag.
+		_, ok := getQuery(t, 0)
+		assert.False(t, ok)
+	})
```

Comment:
I feel this is a breaking change. I'd rather to set a max default length to truncate to than having a case where we don't set the tag.